File: CSES/ProblemSet/IntroductoryProblems/Repetitions/repetitions.c
#include <stdio.h>

int repetitions();

/**
 * Solution to the CSES "Repetition" problem
 * https://cses.fi/problemset/task/1069
 */
int main()
{  
  printf("%d\n", repetitions());
}

/**
 * repetitions: Reads from the console and solves the problem.
 * @return the length of the longest repetition
 */
int repetitions()
{
  char c, aux;
  int i, cont, max;
  
  i = max = cont = 1;
  while((c = getchar()) != EOF){
    if(i == 1){
      aux = c;
      i++;
    }
    else{
      if(aux == c){
	cont++;
	if(cont > max)
	  max = cont;
      }
      else{
	cont = 1;
	aux = c;
      }
    }
  }
  return max;
}

File: CSES/ProblemSet/IntroductoryProblems/Repetitions/repetitions.hs
-- | repetitions: Recursive function, solves the problem.  
-- String, aux, cont, max
repetitions :: [Char] -> Char -> Integer -> Integer -> Integer
repetitions (x:[]) c cont m
  | x == c && m==cont = m+1
  | otherwise = maxx
  where
    maxx
      | m > cont = m
      | otherwise = cont
repetitions (x:xs) c cont m
  | x == c = repetitions xs c cont2 m2
  | x /= c = repetitions xs x 1 m
  where 
    cont2 = cont + 1
    m2
      | m > cont2 = m
      | otherwise = cont2

-- | Solution to CSES "Repetitions" problem.
--   link: https://cses.fi/problemset/task/1069
main :: IO()
main = do
  s <- getLine
  let r = repetitions s (head s) 0 0
  print r
